src/graph.py

```diff
@@ -52,25 +53,27 @@ def edges(self):
         list_key_value = self._dict.items()
         list_edges = []
         for pair in list_key_value:
-            for node in pair[1]:
-                list_edges.append((pair[0], node),)
+            for tup in pair[1]:
+                list_edges.append((pair[0], tup[0], tup[1]),)
         return list_edges
 
     def del_node(self, n):
         """Delete a node from graph."""
-        if (n) in self._dict:
+        if n in self._dict:
             del self._dict[n]
             for key in self._dict:
-                if (n) in self._dict[key]:
-                    node_value = self._dict[key]
-                    node_value.remove(n)
+                for tup in self._dict[key]:
+                    if n in tup:
```

Comment:
Line 66 here might seem like it is a good conditional check, but this will cause a problem in a scenario where both the node and the weight are both an integer, and are the same integer!

e.g. if you are trying to delete node `2` from the graph, and edge `(3,4)` has a weight of 2, you will end up deleting the tuple `(4, 2)` from within the value list belonging to the key `3` in the dictionary.